room_assistance/indico_room_assistance/plugin.py

```diff
@@ -91,3 +97,29 @@ def _extend_services_menu(self, reservation, **kwargs):
 
     def _get_room_assistance_request(self, sender, **kwargs):
         return RoomAssistanceRequest
+
+    def _on_event_update(self, event, **kwargs):
+        changes = kwargs['changes']
+        if not changes.viewkeys() & {'location_data', 'start_dt', 'end_dt'}:
+            return
+
+        request = Request.find_latest_for_event(event, RoomAssistanceRequest.name)
+        if not request or request.state != RequestState.accepted:
+            return
+
+        if 'location_data' in changes and not event_has_room_with_support_attached(event):
+            data = dict(request.data, comment=render_plugin_template('auto_reject_no_supported_room.txt'))
+            request.definition.reject(request, data, User.get_system_user())
+            flash(_("The new event location is not in the list of the rooms supported by the room assistance team. "
+                    "Room assistance request has been rejected and support will not be provided."), 'warning')
+        if changes.viewkeys() & {'start_dt', 'end_dt'}:
+            tz = pytz.timezone(config.DEFAULT_TIMEZONE)
+            req_dates = {dateutil.parser.parse(occ).astimezone(tz).date() for occ in request.data['occurrences']}
+            event_dates = set(event.iter_days())
+            old_dates = req_dates - event_dates
+            if old_dates:
```

Comment:
If there is no overlap between old and new dates I think we should reject the request.